### PR TITLE
Adapt reference guide to new TOC experience

### PR DESCRIPTION
**Pull request by sabbyanandan**
- Bump the local-server version to 1.2.3.BS (version pulled in the docs for includes)
- Add new TOC experience related infrastructure
- Reorder sections to adapt to general core reference guide for consistency
- Fix appendices section

Resolves spring-cloud/spring-cloud-dataflow-server-cloudfoundry#342

### CODE REVIEW

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -27,30 +27,20 @@ endif::backend-html5[]
...
+include::{dataflow-asciidoc}/shell.adoc[]
 include::{dataflow-asciidoc}/streams.adoc[]
-
 ifndef::omit-tasks-docs[]
 include::{dataflow-asciidoc}/tasks.adoc[]
-endif::omit-tasks-docs[]
-
 include::cf-tasks.adoc[]
-
+endif::omit-tasks-docs[]
```
Comment (trisberg) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/index.adoc]: why did you move this? seems like the original intent was to only exclude the task docs from core if `omit-tasks-docs` was set

Comment (sabbyanandan) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/index.adoc]: The goal for this toggle was to show Task related docs only for compatible CF-server releases. I don't see a reason why we need to explain "Tasks on CF" when the core docs are hidden for incompatible releases, so I clubbed them together.

Comment (trisberg) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/index.adoc]: ahh, that makes sense now

**Comment by trisberg** [comment]
LGTM except the server configuration section has no CF specific content - see #344

**Comment by trisberg** [comment]
merged via commit 05ce2724d6b67b17e1a97bdf1a3d97b06a133510
